c_src/Makefile.mak

```diff
@@ -1,9 +1,85 @@
-all:
-	if not exist "priv" mkdir priv
-	MSBuild c_src/erlocisln.sln /t:Build /p:Configuration=$(conf) /p:Platform="x64" /m
-	@echo Build $(conf)
-
-clean:
-	MSBuild c_src/erlocisln.sln /t:Clean /p:Configuration=$(conf) /p:Platform="x64" /m
-	del priv\*.lib priv\*.dll priv\*.exe /f /q
-	@echo Clean $(conf)
+CPP  = "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15.26726\bin\Hostx64\x64\cl.exe"
+LIB  = lib.exe
+LINK = link.exe
+
+
+# ----- VERSION MACROS -----
+# Visual Studio Library usually found in C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\
+VS_LIB_VERSION = 14.15.26726
+
+# version of the Win SDK usually found in C:\Program Files (x86)\Windows Kits\10\Lib\
+WIN_SDK_VERSION = 10.0.17134.0
+# --------------------------
```

Comment:
Is there a way to autodetect these versions (or the path) instead of hardcoding them?